src/main/java/org/shanerx/tradeshop/bitrade/BiShopCreateEventListener.java

```diff
@@ -48,10 +48,11 @@ public BiShopCreateEventListener(TradeShop instance) {
     @SuppressWarnings("deprecation")
     @EventHandler(priority = EventPriority.HIGH)
     public void onSignChange(SignChangeEvent event) {
+        String header = ShopType.BITRADE.header();
         Player player = event.getPlayer();
         Sign s = (Sign) event.getBlock().getState();
 
-        if (!event.getLine(0).equalsIgnoreCase("[BiTrade]")) {
+        if (!(s.getLine(0).equalsIgnoreCase(header))) {
```

Comment:
The variable `s` is an instance of the `Sign`. This means that it refers to the block related to the `SignChangeEvent`. This might sound right, but the event is fired *before* the block is updated. This means that `event.getLine(int)` is **not** equivalent to `s.getLine(int)`.